libs/fs.js

```diff
@@ -521,13 +521,15 @@ var fs = (function() {
       from = buffer.contentSize;
     }
 
-    var newLength = (from + data.byteLength > buffer.contentSize) ? (from + data.byteLength) : (buffer.contentSize);
+    var newLength = (from + length > buffer.contentSize) ? (from + length) : (buffer.contentSize);
 
     buffer.setSize(newLength);
 
-    buffer.array.set(data, from);
+    for (var i = 0; i < length; i++) {
```

Comment:
IIRC @mbebenita run some measurements when self-hosting some Java class. I don't recall what was the best threshold, it should be easy to build a simple JS benchmark to find it.
On install run we're always writing a single byte, so the for loop is definitely faster, but in the general case we don't know how much data we're writing.